drivers/include/bq2429x.h

```diff
@@ -0,0 +1,500 @@
+ *
+ * @pre @p dev != NULL
+ *
+ * @param[in]   dev Device descriptor.
+ *
+ * @return BQ2429X_OK on success.
+ * @return BQ2429X_ERR_I2C on I2C failure.
+ */
+int bq2429x_disable_charge(const bq2429x_t *dev);
+
+/**
+ * @brief   Set Input Voltage Limit.
+ *
+ * @pre @p dev != NULL
+ *
+ * @param[in]   dev     Device descriptor.
+ * @param[in]   vlim    Voltage limit.
+ *
+ * @return BQ2429X_OK on success.
+ * @return BQ2429X_ERR_I2C on I2C failure.
+ */
+int bq2429x_set_vlim(bq2429x_t *dev, bq2429x_input_voltage_limit_t vlim);
+
+/**
+ * @brief   Get Input Voltage Limit.
+ *
+ * @pre @p dev != NULL && @p vlim != NULL
+ *
+ * @param[in]   dev     Device descriptor.
+ * @param[out]  vlim    Voltage limit.
+ *
+ * @return BQ2429X_OK on success.
+ * @return BQ2429X_ERR_I2C on I2C failure.
+ */
+int bq2429x_get_vlim(const bq2429x_t *dev, bq2429x_input_voltage_limit_t *vlim);
+
+/**
+ * @brief   Set Input Current Limit.
+ *
+ * @pre @p dev != NULL
+ *
+ * @param[in]   dev     Device descriptor.
+ * @param[in]   ilim    Current limit.
+ *
+ * @return BQ2429X_OK on success.
+ * @return BQ2429X_ERR_I2C on I2C failure.
+ */
+int bq2429x_set_ilim(bq2429x_t *dev, bq2429x_input_current_limit_t ilim);
+
+/**
+ * @brief   Get Input Current Limit.
+ *
+ * @pre @p dev != NULL && @p ilim != NULL
+ *
+ * @param[in]   dev     Device descriptor.
+ * @param[out]  ilim    Current limit.
+ *
+ * @return BQ2429X_OK on success.
+ * @return BQ2429X_ERR_I2C on I2C failure.
+ */
+int bq2429x_get_ilim(const bq2429x_t *dev, bq2429x_input_current_limit_t *ilim);
+
+/**
+ * @brief   Set Charge Current.
+ *
+ * @pre @p dev != NULL
+ *
+ * @param[in]   dev     Device descriptor.
+ * @param[in]   ichg    Charge current.
+ *
+ * @return BQ2429X_OK on success.
+ * @return BQ2429X_ERR_I2C on I2C failure.
+ */
+int bq2429x_set_ichg(bq2429x_t *dev, bq2429x_charge_current_t ichg);
+
+/**
+ * @brief   Get Charge Current.
+ *
+ * @pre @p dev != NULL && @p ichg != NULL
+ *
+ * @param[in]   dev     Device descriptor.
+ * @param[out]  ichg    Charge current.
+ *
+ * @return BQ2429X_OK on success.
+ * @return BQ2429X_ERR_I2C on I2C failure.
+ */
+int bq2429x_get_ichg(const bq2429x_t *dev, bq2429x_charge_current_t *ichg);
+
+/**
+ * @brief   Set Charge Voltage Limit.
+ *
+ * @pre @p dev != NULL
+ *
+ * @param[in]   dev     Device descriptor.
+ * @param[in]   vreg    Voltage limit.
+ *
+ * @return BQ2429X_OK on success.
+ * @return BQ2429X_ERR_I2C on I2C failure.
+ */
+int bq2429x_set_vreg(bq2429x_t *dev, bq2429x_charge_voltage_limit_t vreg);
+
+/**
+ * @brief   Get Charge Voltage Limit.
+ *
+ * @pre @p dev != NULL && @p vreg != NULL
+ *
+ * @param[in]   dev     Device descriptor.
+ * @param[out]  vreg    Voltage limit.
+ *
+ * @return BQ2429X_OK on success.
+ * @return BQ2429X_ERR_I2C on I2C failure.
+ */
+int bq2429x_get_vreg(const bq2429x_t *dev,
+                     bq2429x_charge_voltage_limit_t *vreg);
+
+/**
+ * @brief   Update charge parameters on the device.
```

Comment:
So this must be called after setting all those limits?

Reply:
Useful only to refresh the parameters on the device if a fault occurs and the information gets lost. So, usually the process is:

1. Fault occurs and is reported through the interrupt.
2. Refresh the parameters on the device.

Reply:
Made this a private static function, since it's only useful during device initialization, and `bq2429x_init` does it anyway.

8354f26baa23c2bf4e41be45fface1224b9d929b